libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/CoreBotAdapter.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Bot.Builder.Integration.Runtime
+{
+    /// <summary>
+    /// Defines the bot runtime standard implementation of <see cref="BotFrameworkHttpAdapter"/>.
+    /// </summary>
+    internal class CoreBotAdapter : BotFrameworkHttpAdapter
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CoreBotAdapter"/> class.
+        /// </summary>
+        /// <param name="services">Services registered with the application.</param>
+        /// <param name="configuration">Application configuration.</param>
+        public CoreBotAdapter(IServiceProvider services, IConfiguration configuration)
+            : base(
+                services.GetService<ICredentialProvider>(),
+                services.GetService<AuthenticationConfiguration>(),
+                services.GetService<IChannelProvider>(),
+                logger: services.GetService<ILogger<BotFrameworkHttpAdapter>>())
+        {
+            var conversationState = services.GetService<ConversationState>();
+            var userState = services.GetService<UserState>();
+
+            this.UseStorage(services.GetService<IStorage>())
+                .UseBotState(userState, conversationState)
+                .Use(new RegisterClassMiddleware<IConfiguration>(configuration));
+
+            // Pick up feature based middlewares: telemetry, inspection, transcripts, etc
+            var middlewares = services.GetServices<IMiddleware>();
+
+            foreach (IMiddleware middleware in middlewares)
+            {
+                this.Use(middleware);
+            }
+
+            OnTurnError = async (turnContext, exception) =>
+            {
+                // Log any leaked exception from the application
+                Logger.LogError(exception, exception.Message);
+
+                // Send the exception message to the user. Since the default behavior does not
+                // send logs or trace activities, the bot appears hanging without any activity
+                // to the user.
+                await turnContext.SendActivityAsync(exception.Message).ConfigureAwait(false);
```

Comment:
Should exceptions from this be caught?

Reply:
Might lead to duplicate error logs if the reason we cannot send is the same reason that the turn failed, but in some cases they might be different and provide more debugging insights. Got a small repro of such scenario so seems like a good idea. Thanks.